pyproject.toml

```diff
@@ -97,49 +95,70 @@ style = "pep440"
 target-version = "py38"
 line-length = 127
 
+[tool.ruff.lint]
 select = [
-    "F",     # Pyflakes
-    "W",     # Pycodestyle (warnigns)
-    "E",     # Pycodestyle (errors)
-    "N",     # pep8-naming
-    "ANN",   # flake8-annotations
-    "B",     # flake8-bugbear
-    "FA",    # flake8-future-annotations
-    "TID",   # flake8-tidy-imports
-    "UP007", # flake8-new-union-types equivalent rule
-    "RUF",   # ruff's custom rules
+  "F",     # Pyflakes
+  "W",     # Pycodestyle (warnigns)
+  "E",     # Pycodestyle (errors)
+  "N",     # pep8-naming
+  "ANN",   # flake8-annotations
+  "B",     # flake8-bugbear
+  "FA",    # flake8-future-annotations
+  "TID",   # flake8-tidy-imports
+  "UP007", # flake8-new-union-types equivalent rule
+  "RUF",   # ruff's custom rules
 ]
 
 ignore = [
-    "ANN002", # Missing type annotation for *args
-    "ANN003", # Missing type annotation for **kwargs
-    "ANN101", # Missing type annotation for self in method
-    "ANN102", # Missing type annotation for cls in classmethod
-    "ANN204", # Missing return type annotation for special method
-    "B904",   # Exception raised within try-except should use raise ... from exc
-    "RUF012", # Mutable class attributes should be annotated with `typing.ClassVar`
+  "ANN002", # Missing type annotation for *args
+  "ANN003", # Missing type annotation for **kwargs
+  "ANN101", # Missing type annotation for self in method
+  "ANN102", # Missing type annotation for cls in classmethod
+  "ANN204", # Missing return type annotation for special method
+  "B904",   # Exception raised within try-except should use raise ... from exc
+  "RUF012", # Mutable class attributes should be annotated with `typing.ClassVar`
+
+  # Redundant rules with ruff-format:
+  "E111",   # Indentation of a non-multiple of 4 spaces
+  "E114",   # Comment with indentation  of a non-multiple of 4 spaces
+  "E117",   # Cheks for over-indented code
+  "D206",   # Checks for docstrings indented with tabs
+  "D300",   # Checks for docstring that use ''' instead of """
+  "Q000",   # Checks of inline strings that use wrong quotes (' instead of ")
+  "Q001",   # Multiline string that use wrong quotes (''' instead of """)
+  "Q002",   # Checks for docstrings that use wrong quotes (''' instead of """)
+  "Q003",   # Checks for avoidable escaped quotes ("\"" -> '"')
+  "COM812", # Missing trailing comma (in multi-line lists/tuples/...)
+  "COM819", # Prohibited trailing comma (in single-line lists/tuples/...)
+  "ISC001", # Single line implicit string concatenation ("hi" "hey" -> "hihey")
+  "ISC002", # Multi line implicit string concatenation
```

Comment:
We can, but disabling these is recommended by ruff. That's because locally, people don't have to run format before check. Also, ruff has an LSP available, which probably doesn't format before checking. Not to mention that the checking actually also has automatic fixing.

With ruff format, we can for example set indentation style to tabs, or set a different indent width. This is obviously unpythonic but the point is that these rules shouldn't affect the ability to set those if we at some point wanted to. That's why ruff suggest that we disable them.